grr/server/grr_response_server/gui/api_plugins/metadata.py

```diff
@@ -333,6 +413,96 @@ def _CreateSchemas(self) -> None:
         else:
           self._CreateSchema(result_type, visiting)
 
+  def _GetDescribedSchema(
+      self,
+      field_descriptor: FieldDescriptor,
+  ) -> Union[SchemaReference, ArraySchema, DescribedSchema]:
+    """Wrap a type schema in a dictionary with a `description` field, if needed.
+
+    This function takes into consideration the fact that the a message field
+    might have a protobuf type that is not completely supported by the OpenAPI
+    Specification (such as `protobuf.oneof` or `protobuf.map`) and wraps the
+    `Reference Object` or the OpenAPI array accordingly in order to include a
+    description of the complete semantics of the type.
+
+    Args:
+      field_descriptor: The protobuf `FieldDescriptor` associated with the
+        target field.
+
+    Returns:
+      If the schema of the field does not require any description to explain the
+      semantics, then a normal schema or reference, else, a dictionary that
+      includes a `description` entry along the `Reference Object` or schema.
+    """
+    if self.schema_objs is None:  # Check required by mypy.
+      raise AssertionError(
+        "The container of OpenAPI type schemas is not initialized."
+      )
+
+    type_name = _GetTypeName(field_descriptor)
+    containing_oneof: OneofDescriptor = field_descriptor.containing_oneof
+    description: str = ""
```

Comment:
Minor nit: I'd expect Mypy static type analyzer to be smart enough (hopefully) to infer that `description` is a string if it gets initialized by assigning a string literal. So, type annotation is likely not needed here.

Reply:
You're right, thank you! Removed the unnecessary type annotation.